app/static/tests/e2e/options.etest.ts

```diff
@@ -351,4 +351,57 @@ test.describe("Options Tab tests", () => {
         await expect((await page.innerText(":nth-match(.parameter-set  .card-header, 1)")).trim()).toBe("Set 1");
         await expect((await page.innerText(":nth-match(.parameter-set  .card-header, 2)")).trim()).toBe("Set 3");
     });
+
+    const fillInAdvancedInputs = async (type: string, advancedSetting: any, index: number) => {
+        switch (type) {
+        case "num": {
+            const input = await advancedSetting.locator("input");
+            await input.fill((index).toString());
+            break;
+        }
+        case "stdf": {
+            const input1 = await advancedSetting.locator(":nth-match(input, 1)");
+            const input2 = await advancedSetting.locator(":nth-match(input, 2)");
+            await input1.fill(`${(index)}`);
+            await input2.fill(`${(-index)}`);
+            break;
+        }
+        case "tag": {
+            const tagInput = await advancedSetting.locator("input");
+            await tagInput.fill((index).toString());
+            await tagInput.press(",");
+            break;
+        }
+        default: break;
+        }
+    };
+
+    const advancedOptions = [
+        { label: "Tolerance", type: "stdf" },
+        { label: "Max steps", type: "num" },
+        { label: "Max step size", type: "num" },
+        { label: "Min step size", type: "stdf" },
+        { label: "Critical times", type: "tag" }
+    ] as const;
+
+    const expectAdvancedSetting = async (page: Page, index: number) => {
+        const advancedSettingPanel = await page.locator("#advanced-settings-panel");
+        await expect(await page.innerText(`:nth-match(#advanced-settings-panel label, ${index + 1})`))
+            .toBe(advancedOptions[index].label);
+        const advancedSetting = await advancedSettingPanel.locator(`:nth-match(.row, ${index + 1})`);
+        await fillInAdvancedInputs(advancedOptions[index].type, advancedSetting, index + 1);
```

Comment:
Could check the run plot is out of date after editing advanced values

Reply:
done, i checked it at the end of the 5 changes, do you want me to check it in the function? we would need to run model every time to reset run plot out of date

Reply:
actually nevermind, i think it is better to check every time anyway, have done that now!